rename feature name to NPM for installing/uninstalling driver.  Must …

derekwbrown opened the pull request:
…match coincident feature from driver merge module

https://github.com/DataDog/datadog-windows-filter/pull/74
### Motivation

simplify configuration for user.

juliogreff committed:
[clcrunner] Allow runners to use different identifiers

We want to allow users to binpack Cluster Checks Runners to a single
node when necessary. Previously, each CLC runner was identifier by the
node name where they resided, so it was not possible to run more than
one per node without causing collisions.

This commit introduces a `clc_runner_id` config
option to allow users to override that now. It's probably not very
useful as a static value, so the intended usage is by injecting the
`DD_CLC_RUNNER_ID` env var in the pod spec as follows:

```
env:
- name: DD_CLC_RUNNER_ID
  valueFrom:
    fieldRef:
      fieldPath: metadata.name
```

derekwbrown committed:
rename feature name to NPM for installing/uninstalling driver.  Must match coincident feature from driver merge module

derekwbrown committed:
rename feature name to NPM for installing/uninstalling driver.  Must match coincident feature from driver merge module

albertvaka commented:
A quick `grep WindowsNP` reveals there's a reference to it in `customactiondata.cpp`

derekwbrown commented:
> A quick `grep WindowsNP` reveals there's a reference to it in `customactiondata.cpp`

doh.  good catch. thanks.

derekwbrown committed:
fix missed search/replace
retrigger

derekwbrown committed:
fix missed search/replace
retrigger

hmahmood committed:
[system-probe] Ignore udp port binding map update already exists error (#7541)

brycekahle committed:
Fix TestPerfBatchStateCleanup to not depend on actual elapsed time (#7549)

Other non-test connections were influencing this test result. Hopefully by making it assert as fast as possible, will fix the flakiness.

nhinsch committed:
Send timeout metric, handle restarts after timeouts, and send metrics when logs are disabled (#7529)

AlexandreYang committed:
Move SNMP auto_conf.yaml from integrations-core to agent repo (#7545)

zippolyte committed:
Get Cloud Foundry application name from Cloud Controller API (#7483)

* Get Cloud Foundry application name from Cloud Controller API

* tests

* reno

* review

* Only store necessary data

* more test

* Update pkg/config/config_template.yaml

* Use atomic

* nits

* fix race

Gui774ume committed:
[runtime-security] Track process credentials updates (#7493)

* track process credentials with commit_creds hook point

* Fix capset tests

* exclude tool dependency from licensing

* remove gocapability from tools.go

lebauce committed:
Handle multierror in runtime security policy check (#7554)

therve committed:
Align fields in struct (#7552)

hush-hush committed:
Revert "Send timeout metric, handle restarts after timeouts, and send metrics when logs are disabled (#7529)"

This reverts commit 55f8b1ee12c92f9925c684d84692ea4d6150e497.

xlucas committed:
Custom resource version for pods (#7449)

AlexandreYang committed:
Remove snmp default conf (#7547)

juliogreff committed:
Redirect gRPC server logs to the Datadog logger (#7553)

Back in 975fd365ee7e8e84a9bf2736333a83244a7b0f8f we did the same thing
for the gRPC clients, but hadn't noticed the same problem occurred with
the server. To be able to use the same solution across both, now we have
a `pkg/util/grpc` package that also became a little smarter to avoid
duplicating logging lines for higher severities.

This has a small change in behavior as well as it supports
`GRPC_GO_LOG_SEVERITY_LEVEL` and `GRPC_GO_LOG_VERBOSITY_LEVEL` env
variables for finer grained control of gRPC logs.

ahmed-mez committed:
[DSD] Use inclusive terminology (#7559)

AlexandreYang committed:
snmp_listener: add loader config (#7514)

raphaelgavache committed:
pkg/trace: enable client tracers to drop auto_drop/p0 spans (#7422)

* Add client p0 drops

brycekahle committed:
[system-probe] Fix, document, and validate system_probe_config.sysprobe_socket config on windows (#7510)

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

Commit:
Improve resiliency external metrics server Part 1 (#7517)

* Updating import path for WPA v1alpha1
* Migrating core logic to dynamic client and generic Unstructured
* Update deps, licences and tests

albertvaka committed:
Do not override iot_host, just change the default (#7544)

We want to change the default in the iot-agent build, not to override the
value.

derekwbrown committed:
update to latest build with fix and signed driver

derekwbrown committed:
update to latest build with fix and signed driver

derekwbrown committed:
re-add NPM flag for easier install.
Changes component name to NPM.
make main component mandatory so it can't be removed via gui

derekwbrown committed:
Merge branch 'db/rename-feature-npm' of github.com:DataDog/datadog-agent into db/rename-feature-npm

derekwbrown commented:
somehow botched rebase; will open new PR
